xla/service/gpu/tests/reduce_column_layout_change.hlo

```diff
@@ -62,73 +62,86 @@ ENTRY kernel_entry {
 // CHECK:         %tile_origin.0 = mul i32 %[[VAL_27]], 1
 // CHECK:         %tile_origin.1 = mul i32 %[[VAL_26]], 4096
 // CHECK:         %tile_origin.2 = mul i32 %[[VAL_24]], 32
-// CHECK:         store i32 %thread.id.1, ptr %[[VAL_12]], align 4
+// CHECK:         store i32 %thread.id.1, ptr{{( addrspace\(5\))?}} %[[VAL_12]], align 4
 // CHECK:         br label %[[VAL_29:.*]]
 // CHECK:       loop1.loop_header:                                ; preds = %[[VAL_30:.*]], %[[VAL_17]]
-// CHECK:         %[[VAL_31:.*]] = load i32, ptr %[[VAL_12]], align 4
+// CHECK:         %[[VAL_31:.*]] = load i32, ptr{{( addrspace\(5\))?}} %[[VAL_12]], align 4
 // CHECK:         %[[VAL_32:.*]] = icmp uge i32 %[[VAL_31]], %tile_bound.1
 // CHECK:         br i1 %[[VAL_32]], label %[[VAL_33:.*]], label %[[VAL_34:.*]]
 // CHECK:       loop1.loop_body:                                  ; preds = %[[VAL_29]]
 // CHECK:         %[[VAL_35:.*]] = add nuw nsw i32 %[[VAL_31]], 32
-// CHECK:         store i32 %[[VAL_35]], ptr %[[VAL_12]], align 4
+// CHECK:         store i32 %[[VAL_35]], ptr{{( addrspace\(5\))?}} %[[VAL_12]], align 4
 // CHECK:         %[[VAL_36:.*]] = icmp eq i32 %[[VAL_31]], %thread.id.1
-// CHECK:         store i32 0, ptr %[[VAL_11]], align 4
+// CHECK:         store i32 0, ptr{{( addrspace\(5\))?}} %[[VAL_11]], align 4
 // CHECK:         br label %[[VAL_37:.*]]
 // CHECK:       loop2.loop_header:                                ; preds = %[[VAL_38:.*]], %[[VAL_34]]
-// CHECK:         %[[VAL_39:.*]] = load i32, ptr %[[VAL_11]], align 4
+// CHECK:         %[[VAL_39:.*]] = load i32, ptr{{( addrspace\(5\))?}} %[[VAL_11]], align 4
 // CHECK:         %[[VAL_40:.*]] = icmp uge i32 %[[VAL_39]], 32
 // CHECK:         br i1 %[[VAL_40]], label %[[VAL_30]], label %[[VAL_41:.*]]
 // CHECK:       loop2.loop_body:                                  ; preds = %[[VAL_37]]
 // CHECK:         %[[VAL_42:.*]] = add nuw nsw i32 %[[VAL_39]], 32
-// CHECK:         store i32 %[[VAL_42]], ptr %[[VAL_11]], align 4
+// CHECK:         store i32 %[[VAL_42]], ptr{{( addrspace\(5\))?}} %[[VAL_11]], align 4
 // CHECK:         %[[VAL_43:.*]] = icmp eq i32 %[[VAL_39]], 0
 // CHECK:         %[[VAL_44:.*]] = add i32 %[[VAL_39]], %thread.id.2
 // CHECK:         %[[VAL_45:.*]] = icmp ult i32 %[[VAL_44]], 32
 // CHECK:         br i1 %[[VAL_45]], label %[[VAL_46:.*]], label %[[VAL_38]]
 // CHECK:       x_in_tile-after:                                  ; preds = %[[VAL_46]], %[[VAL_41]]
-// CHECK:         br label %[[VAL_37]], !llvm.loop !5
+// CHECK:         br label %[[VAL_37]], !llvm.loop !{{[0-9]+}}
```

Comment:
maybe separate lines CHECK-GCN and CHECK-PTX?

Reply:
There does not seem to be any value in checking that !5 (I think it's a metadata index). Might be better just to drop it entirely, and end the line at !llvm.loop. 

BTW this is one of the most common divergences between GCN and PTX in these tests, I've seen it in several other files.

Reply:
Yes, ending the line early sounds also good. I would just like to avoid that the patterns become harder to read.